### PR TITLE
Add docker-compose

### PR DESCRIPTION
**Pull request by jordy25519**
A little docker compose config useful for testing a local Nikau

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    - ./data/node-0:/mnt/data
+    command:
+      - --chain=/cennznet/genesis/nikau.raw.json
+      - --validator
+      - --alice
+      - --base-path=/mnt/data
+      - --name=alice-${HOSTNAME:-localhost}-0
+      - --unsafe-ws-external
+      - --unsafe-rpc-external
+      - --rpc-cors=all
```
Comment (KarishmaBothara) [inline docker-compose.yml:15]: just a doubt - why we don't need **node-key** here, we have it for all other service?

Comment (jordy25519) [reply docker-compose.yml:15]: It sets the node's network level public key. It's mostly important for hardcoding bootnodes here

**Review by KarishmaBothara — APPROVED** [review]

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  node-0:
+    image: cennznet/cennznet:1.1.1
+    volumes:
+    - ./data/node-0:/mnt/data
+    command:
+      - --chain=/cennznet/genesis/nikau.raw.json
+      - --validator
+      - --alice
+      - --base-path=/mnt/data
+      - --name=alice-${HOSTNAME:-localhost}-0
```
Comment (aliXsed) [inline docker-compose.yml:12]: Do we need -0, -1 and etc at the end of the names?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
+version: '3'
+services:
+  node-0:
+    image: cennznet/cennznet:1.1.1
```
Comment (aliXsed) [inline docker-compose.yml:4]: Is it temporary that most of them will run the previous version?

File: docker-compose.yml (hunk 2 above)
Comment (jordy25519) [reply docker-compose.yml:12]: Yeah it's the docker container names, it could be named something else if we wanted

File: docker-compose.yml (hunk 3 above)
Comment (jordy25519) [reply docker-compose.yml:4]: Yeah, I'm thinking we'll update this file with each release. For now I was using it to test upgrades from 1.1.1 -> 1.2.0

**Review by aliXsed — APPROVED** [review]

**Comment by jordy25519** [comment]
included in #320

**Pull request closed without merging**